source/common/http/websocket/ws_handler_impl.h

```diff
@@ -0,0 +1,56 @@
+#pragma once
+
+#include <string>
+
+#include "envoy/http/filter.h"
+#include "envoy/network/connection.h"
+#include "envoy/network/filter.h"
+#include "envoy/upstream/cluster_manager.h"
+#include "envoy/upstream/upstream.h"
+
+#include "common/filter/tcp_proxy.h"
+#include "common/network/filter_impl.h"
+
+namespace Envoy {
+namespace Http {
+namespace WebSocket {
+
+/**
+ * An implementation of a WebSocket proxy based on TCP proxy. This filter will instantiate a
+ * new outgoing TCP connection using the defined load balancing proxy for the configured cluster.
+ * All data will be proxied back and forth between the two connections, without any knowledge of
+ * the underlying WebSocket protocol.
+ *
+ * N.B. This class implements Network::ReadFilter interfaces purely for sake of consistency
+ * with TcpProxy filter. WsHandlerImpl it is not used as a network filter in any way.
```

Comment:
I think this entire comment should go. Its outdated and no longer applicable. Sorry about that.